### PR TITLE
Resolving model name in show-budget.

### PR DESCRIPTION
**Pull request by alesstimec**

### CODE REVIEW

**Comment by tasdomas** [comment]
LGTM

File: cmd/showbudget/show_budget.go [hunk 1]
```diff
@@ -69,18 +72,41 @@ func (c *showBudgetCommand) Run(ctx *cmd.Context) error {
...
-	api, err := newAPIClient(client)
+	api, err := newBudgetAPIClient(client)
 	if err != nil {
 		return errors.Annotate(err, "failed to create an api client")
 	}
 	budget, err := api.GetBudget(c.budget)
 	if err != nil {
 		return errors.Annotate(err, "failed to retrieve the budget")
 	}
+	err = c.resolveModelNames(budget)
```
Comment (mattyw) [inline cmd/showbudget/show_budget.go]: I wonder if we should fall back to uuids if model names can't be resolved?

Comment (cmars) [inline cmd/showbudget/show_budget.go]: We should.

Comment (alesstimec) [inline cmd/showbudget/show_budget.go]: doing

File: cmd/showbudget/show_budget.go [hunk 2]
```diff
@@ -69,19 +75,43 @@ func (c *showBudgetCommand) Run(ctx *cmd.Context) error {
...
-	api, err := newAPIClient(client)
+	api, err := newBudgetAPIClient(client)
 	if err != nil {
 		return errors.Annotate(err, "failed to create an api client")
 	}
 	budget, err := api.GetBudget(c.budget)
 	if err != nil {
 		return errors.Annotate(err, "failed to retrieve the budget")
 	}
+	c.resolveModelNames(budget)
```
Comment (tasdomas) [inline cmd/showbudget/show_budget.go]: I would have logged the error from resolveModelNames here, but that works too, I guess.

File: cmd/showbudget/show_budget_test.go [hunk 3]
```diff
@@ -19,16 +21,19 @@ var _ = gc.Suite(&showBudgetSuite{})
...
 	s.stub = &testing.Stub{}
-	s.mockAPI = &mockapi{s.stub}
-	s.PatchValue(showbudget.NewAPIClient, showbudget.APIClientFnc(s.mockAPI))
+	s.mockBudgetAPI = &mockBudgetAPI{s.stub}
+	s.mockAPI = &mockAPI{s.stub}
+	s.PatchValue(showbudget.NewBudgetAPIClient, showbudget.BudgetAPIClientFnc(s.mockBudgetAPI))
+	s.PatchValue(showbudget.NewAPIClient, showbudget.NewAPIClientFnc(s.mockAPI))
 }
 
 func (s *showBudgetSuite) TestShowBudgetCommand(c *gc.C) {
```
Comment (tasdomas) [inline cmd/showbudget/show_budget_test.go]: Add a test where resolving model names returns an error.
Also a test where some models have no info would be good.

**Comment by tasdomas** [comment]
Needs a few more tests. Then LGTM

File: cmd/showbudget/show_budget_test.go (hunk 3 above)
Comment (alesstimec) [inline cmd/showbudget/show_budget_test.go]: done.

**Comment by cmars** [comment]
LGTM

**Comment by alesstimec** [comment]
$$merge$$

**Pull request merged**